core/yaml.adoc

```diff
@@ -0,0 +1,125 @@
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+WildFly will be able to take a list of YAML files from the command line to redefine its exisiting configuration.
+The operations will be applied in the order the files are defined and after the initial operations defined by the XML configuration.
+The YAML configuration should be *idempotent* so that we can start and restart with the same command line without failure nor resulting with a different final configuration.
+The YAML structure should follow the management model structure. We may use YAML local tags to support operations syntax.
+So at boot, WildFly will load and parse its XML configuration, then add all the required extensions. It will then parse the YAML files, looking for a `wildfly-configuration` element as root fo the configuration updates.
+For each entry, it will check if the resource is already created by the XML, otherwise it will add a new operation to the list of boot operations to add the required resource.
+If the resource already exist then it will add a new operation to the list of boot operations to update the attributes.
+
+=== Hard Requirements
+
+* Support for multiple files.
+External files will take precedence over the XML configuration ones. Then order will follow the command line.
+
+* If the resource is already defined then all we have to du is updating its attrbiutes using the `write-attribute` operation.
+* If the YAML part define an empty resource and the resource already exists: log a WARNING, otherwise try to add it.
+* If the YAML is incorrect: like defining an non-existing attribute or child resource we should fail.
+
+=== Nice-to-Have Requirements
+
+* Undefine attribute: using a YAML tag !undefine. For example:
+----
+wildfly-configuration:
+    subsystem:
+        logging:
+          console-handler:
+            CONSOLE: 
+              named-formatter: COLOR_PATTERN
+              level: !undefine
+----
+
+* Allow setting null if value is empty and attribute is null-significant.
+* Allow support to remove resource using a YAML tag !remove. For example:
+----
+wildfly-configuration:
+    subsystem:
+        microprofile-jwt-smallrye: !remove
+----
+
+* Allow support for adding element to list and supporting index: this might impact idempotence.
+
+=== Non-Requirements
+
+The goal of the YAML files is to be able to customize an existing configuration. It is not here to replace the existing configuration support with XML.
```

Comment:
do we want to constraint the management subtrees that can be impacted by this feature:

* [ ] core-service
* [ ] deployment
* [ ] deployment-overlay
* [ ] extension
* [ ] interface
* [ ] path
* [ ] socket-binding-group
* [ ] subsystem
* [ ] system-property

Reply:
Good point

Reply:
I updated the doc on this